### PR TITLE
LPS-172037: Create validators for create and update nested entities

### PR DESCRIPTION
**Pull request by sergiojimcos**

### CODE REVIEW

File: modules/apps/object/object-api/src/main/java/com/liferay/object/related/models/NestedEntityObjectRelatedModelsProviderRegistry.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.related.models;
+
+/**
+ * @author Sergio Jiménez del Coso
+ */
+public interface NestedEntityObjectRelatedModelsProviderRegistry {
```
Comment (ccorreagg) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/related/models/NestedEntityObjectRelatedModelsProviderRegistry.java:20]: I don't understand why we are naming "RelatedModelsProvider" if we are not providing any related model

File: modules/apps/object/object-api/src/main/java/com/liferay/object/related/models/NestedEntityObjectRelatedModelsProvider.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+package com.liferay.object.related.models;
+
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.model.ObjectRelationship;
+
+/**
+ * @author Sergio Jiménez del Coso
+ */
+public interface NestedEntityRelatedModelsProvider {
```
Comment (ccorreagg) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/related/models/NestedEntityObjectRelatedModelsProvider.java]: I don't understand why we are naming "RelatedModelsProvider" if we are not providing any related model

File: modules/apps/object/object-api/src/main/java/com/liferay/object/related/models/NestedEntityObjectRelatedModelsProvider.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	public String getClassName();
+
+	public String getObjectRelationshipType();
+
+	public void validate(
+		Object propertyValue, ObjectEntry serviceBuilderObjectEntry,
```
Comment (ccorreagg) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/related/models/NestedEntityObjectRelatedModelsProvider.java:30]: That is valid when we want to relate an ObjectEntry with another element, but is this be reusable for system objects?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/NestedEntityObjectEntry1toMRelatedModelsProviderImpl.java [hunk 4]
```diff
@@ -18,52 +18,49 @@
...
 
-		_objectDefinition = objectDefinition;
-		_objectFieldLocalService = objectFieldLocalService;
+		super(objectDefinition);
 	}
 
 	@Override
 	public String getClassName() {
-		return _objectDefinition.getClassName();
+		return objectDefinition.getClassName();
```
Comment (ccorreagg) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/NestedEntityObjectEntry1toMRelatedModelsProviderImpl.java:41]: This method is exactly the same in both implementations, could it be moved to the base class?

**Comment by sergiojimcos** [comment]
ci:close

**Pull request closed without merging**